python/dgl/data/adapter.py

```diff
@@ -68,7 +69,10 @@ def __init__(self,
         self.g = dataset[0].clone()
         self.split_ratio = split_ratio
         self.target_ntype = target_ntype
-        self.num_classes = dataset.num_classes
+        if hasattr(dataset, 'num_classes'):
+            self.num_classes = dataset.num_classes
+        else:
+            self.num_classes = len(F.unique(self.g.nodes[self.target_ntype].data['label']))
```

Comment:
Please put it in the `process` method after the  check for `'label'` ndata.

Reply:
fixed